src/main/java/graphql/nadel/engine/NadelExecutionStrategy.java

```diff
@@ -219,28 +211,67 @@ public NadelExecutionStrategy(List<Service> services,
                             return serviceExecutionHooks.resultRewrite(resultRewriteParams);
                         });
 
-
                 return serviceResult;
             }));
         }
         return resultNodes;
     }
 
-    private RootExecutionResultNode getSkippedServiceCallResult(NadelContext nadelContext, ExecutionStepInfo esi, ExecutionContext newExecutionContext, GraphQLError error) {
-        Map<String, Object> errorMap = error.toSpecification();
-
-        HashMap<String, Object> dataMap = new LinkedHashMap<>();
-        String topLevelFieldName = esi.getFieldDefinition().getName();
-        dataMap.put(topLevelFieldName, null);
+    /**
+     * A top level field error is present if the field should not be executed and an
+     * error should be put in lieu. We check this before calling out to the underlying
+     * service. This error is usually present when the field has been rejected by
+     * {@link ServiceExecutionHooks#isFieldForbidden(NormalizedQueryField, Object)}
+     *
+     * @param esi              the {@link ExecutionStepInfo} for the top level field
+     * @param transformedQuery the query for that specific top level field
+     * @return a {@link GraphQLError} if the field was rejected before, otherwise empty
+     */
+    private Optional<GraphQLError> getTopLevelFieldError(ExecutionStepInfo esi, QueryTransformationResult transformedQuery) {
```

Comment:
can we name that more specific to make clear that it is a forbidden field error.